Patch requires second part for module-customer.
"magento/module-customer": [
    {
        "description": "Fix for iconv issue in customer address validation",
        "filename": "essential/magento/module-customer/patch-iconv-issue-magento-eb34e347.patch",
        "version-constraint": ">=102.1.6,<102.1.7"
    }
],
diff --git a/Model/Attribute/Data/Text.php b/Model/Attribute/Data/Text.php
index c41a65a..e025e0b 100644
--- a/Model/Attribute/Data/Text.php
+++ b/Model/Attribute/Data/Text.php
@@ -20,6 +20,11 @@ class Text extends \Magento\Eav\Model\Attribute\Data\AbstractData
      */
     protected $_string;

+    /**
+     * @var array
+     */
+    private $allowDiacriticsForAttributes;
+
     /**
      * @param \Magento\Framework\Stdlib\DateTime\TimezoneInterface $localeDate
      * @param \Psr\Log\LoggerInterface $logger
@@ -31,7 +36,8 @@ class Text extends \Magento\Eav\Model\Attribute\Data\AbstractData
         \Magento\Framework\Stdlib\DateTime\TimezoneInterface $localeDate,
         \Psr\Log\LoggerInterface $logger,
         \Magento\Framework\Locale\ResolverInterface $localeResolver,
-        \Magento\Framework\Stdlib\StringUtils $stringHelper
+        \Magento\Framework\Stdlib\StringUtils $stringHelper,
+        array $allowDiacriticsForAttributes = []
     ) {
         parent::__construct($localeDate, $logger, $localeResolver);
         $this->_string = $stringHelper;
@@ -80,7 +86,17 @@ class Text extends \Magento\Eav\Model\Attribute\Data\AbstractData
         }

         // if string with diacritics encode it.
-        $value = $this->encodeDiacritics($value);
+        // BEGIN PATCH
+        //$value = $this->encodeDiacritics($value);
+        if (isset($this->allowDiacriticsForAttributes[$attribute->getEntityType()->getEntityTypeCode()])
+            && in_array(
+                $attribute->getAttributeCode(),
+                $this->allowDiacriticsForAttributes[$attribute->getEntityType()->getEntityTypeCode()]
+            )) {
+            // if string with diacritics encode it.
+            $value = $this->encodeDiacritics($value);
+        }
+        // END PATCH

         $validateLengthResult = $this->validateLength($attribute, $value);
         $errors = array_merge($errors, $validateLengthResult);
--
2.39.3 (Apple Git-146)
